File: frontend/public/apps/sanction/scripts/process_masak_csv.py
import os
import json
import datetime

# Setup paths
base_dir = os.path.dirname(os.path.abspath(__file__))
raw_dir = os.path.join(base_dir, '../data/masak_raw')
output_file = os.path.join(base_dir, '../data/masak_data.js')

all_records = []

print(f"Reading CSV files from {raw_dir}...")

if not os.path.exists(raw_dir):
    print(f"Directory not found: {raw_dir}")
    exit(1)

for filename in os.listdir(raw_dir):
    if not filename.endswith('.csv'):
        continue
        
    print(f"Processing {filename}...")
    filepath = os.path.join(raw_dir, filename)
    
    # Try reading with different encodings
    encodings = ['cp1254', 'utf-8', 'latin-1']
    lines = []
    
    for enc in encodings:
        try:
            with open(filepath, 'r', encoding=enc) as f:
                lines = f.readlines()
            print(f"Successfully read with {enc}")
            break
        except UnicodeDecodeError:
            continue
            
    if not lines:
        print(f"Failed to read {filename}")
        continue
        
    # Parse Header
    if len(lines) < 2:
        continue
        
    header_line = lines[0].strip()
    headers = [h.strip().upper() for h in header_line.split(';')]
    
    # Column Mapping
    col_map = {
        'name': -1, 'tckn': -1, 'nationality': -1, 'dob': -1, 'pob': -1,
        'mother': -1, 'father': -1, 'org': -1, 'listType': -1, 'decree': -1
    }
    
    for i, h in enumerate(headers):
        if 'AD-SOYAD' in h or 'GERÇEK/TÜZEL' in h or 'ADI SOYADI' in h or 'UNVANI' in h: col_map['name'] = i
        elif 'TCKN' in h or 'VKN' in h or 'PASAPORT' in h: col_map['tckn'] = i
        elif 'UYRU' in h: col_map['nationality'] = i
        elif 'DOĞUM TARİHİ' in h or 'DOGUM TARIHI' in h: col_map['dob'] = i
        elif 'DOĞUM YERİ' in h or 'DOGUM YERI' in h: col_map['pob'] = i
        elif 'ANNE ADI' in h: col_map['mother'] = i
        elif 'BABA ADI' in h: col_map['father'] = i
        elif 'ÖRGÜT' in h or 'ORGUT' in h or 'BAĞLANTILI' in h: col_map['org'] = i
        elif 'YAPTIRIM TÜRÜ' in h or 'YAPTIRIM TURU' in h: col_map['listType'] = i
        elif 'KARAR SAYISI' in h or 'RESMİ GAZETE' in h or 'RESMI GAZETE' in h: col_map['decree'] = i

    print(f"Column Mapping: {col_map}")
    
    # Process Lines
    for line in lines[1:]:
        parts = line.strip().split(';')
        if len(parts) < 2: continue
        
        # Safe get
        def get_col(idx):
            if idx > -1 and idx < len(parts):
                return parts[idx].strip().replace('"', '')
            return ''
            
        name = get_col(col_map['name'])
        if not name or len(name) < 2: continue
        
        record = {
            'name': name,
            'list': 'TR MASAK (Yurtiçi)',
            'type': 'Individual',
            'score': 0,
            'sourceDetails': '',
            'tckn': '',
            'decreeUrl': ''
        }
        
        details = []
        tckn = get_col(col_map['tckn'])
        if tckn:
            record['tckn'] = tckn
            details.append(f"TCKN/VKN/Pasaport: {tckn}")
            
        nat = get_col(col_map['nationality'])
        if nat: details.append(f"Uyruk: {nat}")
        
        dob = get_col(col_map['dob'])
        if dob: details.append(f"Doğum Tarihi: {dob}")
        
        pob = get_col(col_map['pob'])
        if pob: details.append(f"Doğum Yeri: {pob}")
        
        mother = get_col(col_map['mother'])
        if mother: details.append(f"Anne Adı: {mother}")
        
        father = get_col(col_map['father'])
        if father: details.append(f"Baba Adı: {father}")
        
        org = get_col(col_map['org'])
        if org: details.append(f"Örgüt: {org}")
        
        ltype = get_col(col_map['listType'])
        if ltype: details.append(f"Yaptırım Türü: {ltype}")
        
        decree = get_col(col_map['decree'])
        if decree: details.append(f"Karar/Gazete: {decree}")
        
        record['sourceDetails'] = '\n'.join(details)
        
        # Determine List Name
        if 'IC-DONDURMA' in filename:
            record['list'] = 'TR MASAK - İç Dondurma (FETÖ/PKK/DHKP-C/DEAŞ)'
        elif 'BIRLESMIS-MILLETLER' in filename:
            record['list'] = 'TR MASAK - BMGK Kararları'
        elif 'YABANCI-ULKE' in filename:
            record['list'] = 'TR MASAK - Yabancı Ülke Talepleri'
            
        all_records.append(record)

print(f"Total records processed: {len(all_records)}")

# Write to JS file
js_content = f"// MASAK Data generated from CSVs\n// Last Updated: {datetime.datetime.now().isoformat()}\nwindow.MASAK_DATA = {json.dumps(all_records, ensure_ascii=False, indent=2)};\n"

with open(output_file, 'w', encoding='utf-8') as f:
    f.write(js_content)
    
print(f"Successfully wrote to {output_file}")

File: frontend/public/apps/sanction/scripts/process_masak_csv.js
const fs = require('fs');
const path = require('path');
const { TextDecoder } = require('util');

const rawDir = path.join(__dirname, '../data/masak_raw');
const outputFile = path.join(__dirname, '../data/masak_data.js');

console.log(`Reading CSV files from ${rawDir}...`);

let allRecords = [];

try {
    const files = fs.readdirSync(rawDir).filter(f => f.endsWith('.csv'));

    files.forEach(file => {
        console.log(`Processing ${file}...`);
        const filePath = path.join(rawDir, file);
        const buffer = fs.readFileSync(filePath);

        // Try decoding with windows-1254 (Turkish)
        let content;
        try {
            const decoder = new TextDecoder('windows-1254');
            content = decoder.decode(buffer);
        } catch (e) {
            console.warn(`TextDecoder error for ${file}: ${e.message}. Trying iso-8859-9.`);
            try {
                const decoder = new TextDecoder('iso-8859-9');
                content = decoder.decode(buffer);
            } catch (e2) {
                console.warn(`TextDecoder error 2: ${e2.message}. Fallback to latin1.`);
                content = buffer.toString('latin1');
            }
        }

        const lines = content.split(/\r?\n/);
        if (lines.length < 2) return;

        // Parse Header to find column indices
        const headerLine = lines[0];
        const headers = headerLine.split(';').map(h => h.trim().toUpperCase());

        // Identify columns
        const colMap = {
            name: headers.findIndex(h => h.includes('AD-SOYAD') || h.includes('GERÇEK/TÜZEL') || h.includes('ADI SOYADI') || h.includes('UNVANI')),
            tckn: headers.findIndex(h => h.includes('TCKN') || h.includes('VKN') || h.includes('PASAPORT')),
            nationality: headers.findIndex(h => h.includes('UYRU')),
            dob: headers.findIndex(h => h.includes('DOĞUM TARİHİ') || h.includes('DOGUM TARIHI')),
            pob: headers.findIndex(h => h.includes('DOĞUM YERİ') || h.includes('DOGUM YERI')),
            mother: headers.findIndex(h => h.includes('ANNE ADI')),
            father: headers.findIndex(h => h.includes('BABA ADI')),
            org: headers.findIndex(h => h.includes('ÖRGÜT') || h.includes('ORGUT') || h.includes('BAĞLANTILI')),
            listType: headers.findIndex(h => h.includes('YAPTIRIM TÜRÜ') || h.includes('YAPTIRIM TURU')),
            decree: headers.findIndex(h => h.includes('KARAR SAYISI') || h.includes('RESMİ GAZETE') || h.includes('RESMI GAZETE'))
        };

        console.log(`Column Mapping for ${file}:`, colMap);

        // Process lines
        for (let i = 1; i < lines.length; i++) {
            const line = lines[i].trim();
            if (!line) continue;

            // Simple split by semicolon (assuming no semicolons in fields for now)
            // A better regex split would be needed for quoted fields containing delimiters
            const parts = line.split(';');

            if (parts.length < 2) continue;

            const name = colMap.name > -1 ? parts[colMap.name] : '';
            if (!name || name.length < 2) continue;

            const record = {
                name: name.replace(/"/g, '').trim(),
                list: 'TR MASAK (Yurtiçi)',
                type: 'Individual', // Default
                score: 0, // Initial
                sourceDetails: '',
                tckn: '',
                decreeUrl: ''
            };

            // Build Details HTML
            let detailsHtml = '';

            if (colMap.tckn > -1 && parts[colMap.tckn]) {
                const tcknVal = parts[colMap.tckn].trim();
                if (tcknVal) {
                    record.tckn = tcknVal;
                    detailsHtml += `TCKN/VKN/Pasaport: ${tcknVal}\n`;
                }
            }

            if (colMap.nationality > -1 && parts[colMap.nationality]) {
                detailsHtml += `Uyruk: ${parts[colMap.nationality].trim()}\n`;
            }

            if (colMap.dob > -1 && parts[colMap.dob]) {
                detailsHtml += `Doğum Tarihi: ${parts[colMap.dob].trim()}\n`;
            }

            if (colMap.pob > -1 && parts[colMap.pob]) {
                detailsHtml += `Doğum Yeri: ${parts[colMap.pob].trim()}\n`;
            }

            if (colMap.mother > -1 && parts[colMap.mother]) {
                detailsHtml += `Anne Adı: ${parts[colMap.mother].trim()}\n`;
            }

            if (colMap.father > -1 && parts[colMap.father]) {
                detailsHtml += `Baba Adı: ${parts[colMap.father].trim()}\n`;
            }

            if (colMap.org > -1 && parts[colMap.org]) {
                detailsHtml += `Örgüt: ${parts[colMap.org].trim()}\n`;
            }

            if (colMap.listType > -1 && parts[colMap.listType]) {
                detailsHtml += `Yaptırım Türü: ${parts[colMap.listType].trim()}\n`;
            }

            if (colMap.decree > -1 && parts[colMap.decree]) {
                detailsHtml += `Karar/Gazete: ${parts[colMap.decree].trim()}\n`;
            }

            record.sourceDetails = detailsHtml.trim();

            // Determine List Name based on filename or content
            if (file.includes('IC-DONDURMA')) {
                record.list = 'TR MASAK - İç Dondurma (FETÖ/PKK/DHKP-C/DEAŞ)';
            } else if (file.includes('BIRLESMIS-MILLETLER')) {
                record.list = 'TR MASAK - BMGK Kararları';
            } else if (file.includes('YABANCI-ULKE')) {
                record.list = 'TR MASAK - Yabancı Ülke Talepleri';
            }

            allRecords.push(record);
        }
    });

    console.log(`Total records processed: ${allRecords.length}`);

    // Write to JS file
    const jsContent = `// MASAK Data generated from CSVs
// Last Updated: ${new Date().toISOString()}
window.MASAK_DATA = ${JSON.stringify(allRecords, null, 2)};
`;

    fs.writeFileSync(outputFile, jsContent, 'utf-8');
    console.log(`Successfully wrote to ${outputFile}`);

} catch (e) {
    console.error('Error processing CSVs:', e);
}
